src/algorithms/reco/ElectronReconstruction.cc

```diff
@@ -55,10 +55,14 @@ namespace eicrecon {
               auto reco_part = reco_part_assoc->getRec();
               double EoverP = clu.getEnergy() / edm4hep::utils::magnitude(reco_part.getMomentum());
               m_log->trace( "ReconstructedParticle: Energy={} GeV, p={} GeV, E/p = {} for PDG (from truth): {}", clu.getEnergy(), edm4hep::utils::magnitude(reco_part.getMomentum()), EoverP, sim.getPDG() );
+              auto reco_electron = reco_part.clone();
+              reco_electron.addToClusters( clu );
```

Comment:
Two questions:
- We are cloning here instead of reusing the reconstructed electron and storing it in a subset collection. Is that what we want? If we think of the ElectronReconstruction algorithm as following the cluster matching step, then electron finding should not change the electron and a clone should not be necessary, right?
- Relatedly, why isn't `reco_part` not yet filled with clusters? Shouldn't we be using the `ReconstructedParticles`  (tracking + PID + cluster matching) as input here instead of the `ReconstructedChargedParticles` (which are from tracking + PID). We seem to be repeating the cluster matching that is being done in `MatchClusters` here.

Reply:
Hi wouter, both good points:
- When I first implemented this we did not have the option for subset collections. Is there an example I can follow of a working subset collection? If so I can test that and modify this case.
- I did not realize the difference - I will try modifying this to use `ReconstructedParticles` which should also simplify the code

Reply:
I think the main active example of the subset collection is the TrackerHitCollector: https://github.com/eic/EICrecon/blob/main/src/algorithms/tracking/TrackerHitCollector.h#L36.

Reply:
Should we merge this and file an issue for refactoring?